### ISSUE TITLE
ERROR [099] - raised from local protocol: invalid type at

### ISSUE THREAD

**Issue by Kobaruon** [pgbackrest/pgbackrest#1348]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.31

2. PostgreSQL version: 12.3 and 11.5

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Both of my database servers and backup servers are on CentOS 7.9.

4. Did you install pgBackRest from source or from a package? Package

5. Please attach the following as applicable:
pgbackrest log:
```
2021-03-26 02:40:18.067 P01   INFO: backup file 10.250.20.122:/pgsql-data/pg_logical/replorigin_checkpoint (8B, 0%) checksum 347fc8f2df71bd4436e38bd1516ccd7ea0d46532
2021-03-26 02:40:18.278 P01   INFO: backup file 10.250.20.122:/pgsql-data/PG_VERSION (3B, 0%) checksum ad552e6dc057d1d825bf49df79d6b98eba846ebe
ERROR: [099]: raised from local-3 protocol: invalid type at 'η��F�T��J}<�{�����&'
WARN: local-2 process did not exit when expected
WARN: local-3 process did not exit when expected
WARN: local-4 process did not exit when expected
2021-03-26 04:11:49.291 P00   INFO: backup command end: aborted with exception [099]
```
```
2021-03-25 03:56:04.928 P00   INFO: backup start archive = 0000000E0000023700000003, lsn = 237/3000028
ERROR: [099]: raised from local-4 protocol: invalid type at '�w�^/�h̋xǲ�+�y"���&�����ud���5Ec�wx$�<f���P��s�O��-yQ4�E<���8�������
��Eј�ޚM^3vݥ��\����OՉ<(� ^�@��OH']�d��j�Ec^�g�]����LxkR�Lј�z���fu�'
WARN: local-1 process did not exit when expected
WARN: local-3 process did not exit when expected
2021-03-25 04:57:08.681 P00   INFO: backup command end: aborted with exception [099]
```

postgresql.log
```
< user=postgres db=postgres host=[local] pid=46729 time=2021-03-26 11:07:27.294 +03 app=pgBackRest [backup] > WARNING:  aborting backup due to backend exiting before pg_stop_backup was called
```

7. Describe the issue: I have an issue on two database clusters that are completely different versions (12.3 and 11.5). I use pgbackrest 2.31 on both of them. They were working fine for the last 2 months. For some reason, both of them started to give this error.
I have more than 30 clusters that works with pgbackrest 2.31, only these are problematic.

**Comment by dwsteele** [comment]
I don't immediately see what is going on here -- it could be network or I/O errors.

It would be helpful if you would provide your configuration files. Also, let's start by getting debug output of a failed backup by adding `--log-level-file=debug`. Is the failure consistent or intermittent?

**Comment by Kobaruon** [comment]
The failure is consistent on one cluster. However, it sometimes works on the other cluster. Also, as I said before we have over 50 cluster that are working fine with this configuration.

You can find the debug log here.
[pgbackrest.log](https://github.com/pgbackrest/pgbackrest/files/6221995/pgbackrest.log)

On backup server;
pgbackrest.conf
```
[global]
repo1-path=/pgsql-backup
repo1-host-user=postgres
process-max=4
log-level-console=info
log-level-file=detail

[pg_cluster]
pg1-host=node1
pg1-host-user=postgres
pg1-path=/pgsql-data
pg1-port=5432
pg2-host=node2
pg2-host-user=postgres
pg2-path=/pgsql-data
pg2-port=5432
pg3-host=node3
pg3-host-user=postgres
pg3-path=/pgsql-data
pg3-port=5432
backup-standby=y
compress=y
repo1-retention-full=1
start-fast=y
stop-auto=y
```

On cluster nodes;
pgbackrest.conf
```
[global]
repo1-host= backup_server_ip
repo1-host-user=postgres
log-level-console=info
log-level-file=detail
compress=n

[global:archive-push]
process-max=1
archive-async=y
log-level-console=info
log-level-stderr=info

[pg_cluster]
pg1-path=/pgsql-data
pg1-port=5432
```

**Comment by Kobaruon** [comment]
I tried with --log-subprocess and debug. This is the actual error log.
```
2021-03-31 08:41:05.027 P00  DEBUG:     command/backup/backup::backupJobResult: (manifest: {Manifest}, host: {"node2"}, fileName: {"/pgsql-data/base/16388/87833"}, fileRemove: {[]}, job: {state: done, key: {"pg_data/base/16388/87833"}, command: {command: backupFile}, code: 99, message: {"raised from local-4 protocol: invalid type at '>Vd�2r0��e|�x�'
```

This is the `/var/log/pgbackrest/pg_cluster-backup-local-004.log` file. Sadly, it did not do anything as far as I understand.
```
-------------------PROCESS START-------------------
2021-03-31 08:41:03.362 P00   INFO: backup:local command begin 2.31: --backup-standby --checksum-page --compress --exec-id=153907-735ef167 --log-level-console=off --log-level-file=detail --log-level-stderr=error --log-subprocess --pg=2 --pg1-host=node1 --pg2-host=node2 --pg3-host=node3 --pg1-host-user=postgres --pg2-host-user=postgres --pg3-host-user=postgres --pg1-path=/pgsql-data --pg2-path=/pgsql-data --pg3-path=/pgsql-data --pg1-port=5432 --pg2-port=5432 --pg3-port=5432 --process=4 --process-max=4 --remote-type=pg --repo1-path=/pgsql-backup --repo1-retention-full=1 --stanza=pg_cluster --start-fast --stop-auto --type=full
```

We've upgraded most of our clusters recently. Could this be the reason? But, not all clusters that are upgraded give this error only couple of them do. The upgrades were minor like 11.5>11.11 or 12.3>12.6. We did not run pg_upgrade as that's not necessary.

**Comment by dwsteele** [comment]
Hmmm, how about we try that again at `trace` level?

Did anything else get updated when postgres was update? Kernel, libs, etc.?

**Comment by Kobaruon** [comment]
I executed the backup with trace flag. This is the local protocol log that gives an error.
[pg_cluster-backup-local-003.log](https://github.com/pgbackrest/pgbackrest/files/6249635/pg_cluster-backup-local-003.log)

Complete backup log was over 50mb so I had to compress it.
[pg_cluster-backup.log.zip](https://github.com/pgbackrest/pgbackrest/files/6249674/pg_cluster-backup.log.zip)

**Comment by dwsteele** [comment]
OK, looks like this is the issue:
```
2021-04-02 15:52:45.726 P00   WARN: 'compress' and 'compress-type' options should not both be set
                                    HINT: 'compress-type' is preferred and 'compress' is deprecated.
```
We should be changing the deprecated `compress` to `compress-type` internally but looks like there may be an issue there. We'll look into it. I think it would still be fixed in the next release and since we suppressed warnings to fix a related bug.

But in the meantime, you can should all references to `compress` and use `compress-type` instead. I think that will solve the problem.

**Comment by Kobaruon** [comment]
I changed `compress` to `compress-type=none` however the issue still remains.

Again here are the trace level logs

[pg_cluster-backup-local-002.log](https://github.com/pgbackrest/pgbackrest/files/6256675/pg_cluster-backup-local-002.log)
[pg_cluster-backup.log 2.zip](https://github.com/pgbackrest/pgbackrest/files/6256676/pg_cluster-backup.log.2.zip)

**Comment by dwsteele** [comment]
Ok, on the `node1` host there should be a log file named `pg_cluster-backup-remote-002.log`. Let's have a look at that log.

**Comment by Kobaruon** [comment]
Okay, these are the logs on node1
[pg_cluster-backup-remote-002.log](https://github.com/pgbackrest/pgbackrest/files/6258494/pg_cluster-backup-remote-002.log)

**Comment by dwsteele** [comment]
I've been through all the logs and I don't see anything unusual except that garbage is being read off the SSH connection from the local to the remote. I am not able to reproduce this issue and I'm not sure what other info to collect. I'll let you know if I think of anything, but if you have many servers that work and one that doesn't then you'll want to look at what makes this server different.

Different configuration? Bigger cluster? OS Version? You said this server was working until you updated Postgres. Was anything else updated that time?

**Comment by Kobaruon** [comment]
Apart from updating Postgres, we installed python3 and updated patroni, but they should not interfere with anything as far as I know. Just like you said, we have many clusters that are working with the same pgbackrest and postgres version. 

Also, we had 4 clusters that were giving this same error. I removed some of the standby nodes from `pgbackrest.conf` and they worked. However, in this cluster every node gives this error so I am not sure what's wrong with this one.

It's not really a big cluster <100GB, also same OS and configuration.
